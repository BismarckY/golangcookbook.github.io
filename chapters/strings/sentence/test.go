package main

import (
	"fmt"
	"strings"
)

func ToSentence(words []string, andOrOr string) string {
	l := len(words)
	wordsForSentence := make([]string, l)
	copy(wordsForSentence, words)
	wordsForSentence[l-1] = andOrOr + " " + wordsForSentence[l-1]
	return strings.Join(wordsForSentence, ", ")
}

func main() {
	a := []string{"this", "that", "the other"}
	fmt.Println(strings.Join(a, ", "))
	fmt.Println(ToSentence(a, "or"))
}
