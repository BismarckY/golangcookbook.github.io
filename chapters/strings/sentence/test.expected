this, that, the other
this, that, or the other
